Mypy reports an error with Numpy 1.21

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Information

- **Qiskit Optimization version**: latest
- **Python version**: python 3.7.10, python 3.8.5
- **Operating system**: ubuntu 20.04.2

### What is the current behavior?

`make mypy` reports the following error with the main branch.
```
mypy qiskit_optimization test tools
qiskit_optimization/algorithms/admm_optimizer.py:685: error: Incompatible types in assignment (expression has type "ndarray[Any, Any]", variable has type "QuadraticExpression")
Found 1 error in 1 file (checked 99 source files)
```

This happens with numpy 1.21 (released 3 hours ago). https://github.com/numpy/numpy/releases/tag/v1.21.0
Mypy does not report the error with numpy 1.20.

The following line causes the error. Mypy cannot perhaps handle setter `quadratic` properly.
https://github.com/Qiskit/qiskit-optimization/blob/7f341b00d850577cf866caecbee61a102144156e/qiskit_optimization/algorithms/admm_optimizer.py#L685
Interestingly, a similar code (setter of `linear`) is treated as no error. I cannot see any difference between them.
https://github.com/Qiskit/qiskit-optimization/blob/7f341b00d850577cf866caecbee61a102144156e/qiskit_optimization/algorithms/admm_optimizer.py#L689

This may be related to https://github.com/python/mypy/issues/3004

### Steps to reproduce the problem

`make mypy` with numpy 1.21.

### What is the expected behavior?

no error

### Suggested solutions

Shall we add `# type: ignore`?

--- a/qiskit_optimization/algorithms/admm_optimizer.py
+++ b/qiskit_optimization/algorithms/admm_optimizer.py
@@ -682,7 +682,7 @@ class ADMMOptimizer(OptimizationAlgorithm):
         quadratic_y = self._params.beta / 2 * np.eye(binary_size) + self._state.rho / 2 * np.eye(
             binary_size
         )
-        op3.objective.quadratic = quadratic_y
+        op3.objective.quadratic = quadratic_y  # type: ignore[assignment]
 
         # set linear objective for y
         linear_y = -self._state.lambda_mult - self._state.rho * (self._state.x0 - self._state.z)

